### PR TITLE
Drop share

### LINKED ISSUE #88: drop_shared
Relevant URLs

```
share_url <- "https://api.dropbox.com/1/shares/auto/"
```

Notes from @scheidec

- Code: https://github.com/karthik/rdrop2/blob/master/R/drop_file_ops.R
- Change from v1: /1/shares to v2: /2/sharing/create_shared_link_with_settings
- Parameters and return values have changed
- V1 docs: https://www.dropbox.com/developers-v1/core/docs#shares
- V2 docs: https://www.dropbox.com/developers/documentation/http/documentation#sharing-create_shared_- link_with_settings
‘locale’ and ‘short_url’ are no longer parameters, now just ‘path’ and ‘settings’ are parameters

### PR DESCRIPTION
**Pull request by karthik**
- This PR adds drop_share.
- Now there is also a new function to list all current shares
- Sharing can be public, private (pro accounts only) or team_only (business accounts only)
- This PR also refactors out all use of assertthat and replaces it with assertive.

Closes #88.

### CODE REVIEW

File: R/drop_shared.R [hunk 1]
```diff
@@ -1,26 +1,94 @@
...
+  visibilities <- c("public", "team_only", "password")
+  assertive::assert_any_are_matching_fixed(visibilities, requested_visibility)
+
+  # TODO
+  # Once the new drop_exists is done, one must check to see if a file/folder
+  # exists on Dropbox before proceeding
+
+  path <- add_slashes(path)
+  settings <-
+    drop_compact(
```
Comment (ClaytonJY) [inline R/drop_shared.R]: is this necessary? will `encode = "json"` not handle depth properly?

File: R/drop_shared.R [hunk 2]
```diff
@@ -1,26 +1,94 @@
...
+#' This function returns a list of all links that are currently being shared
+#' @template token
+#' @param verbose Print verbose output
+#'
+#' @export
+#'
+#' @examples \dontrun{
+#' list_shared_links()
+#' }
+list_shared_links <-
```
Comment (ClaytonJY) [inline R/drop_shared.R]: I think this should be prefixed with `drop_` for consistency

File: R/drop_shared.R [hunk 3]
```diff
@@ -1,26 +1,94 @@
...
+        expires = expires
+      )
+    )
+  # settings <- jsonlite::toJSON(settings, auto_unbox = TRUE)
+  share_url <-
+    "https://api.dropboxapi.com/2/sharing/create_shared_link_with_settings"
+
+  req <- httr::POST(
+    url = share_url,
+    httr::config(token = get_dropbox_token()),
```
Comment (ClaytonJY) [inline R/drop_shared.R]: `token = dtoken`

File: R/drop_shared.R (hunk 2 above)
Comment (karthik) [reply R/drop_shared.R]: fixed

File: R/drop_shared.R (hunk 3 above)
Comment (karthik) [reply R/drop_shared.R]: fixed

File: R/drop_shared.R (hunk 1 above)
Comment (karthik) [reply R/drop_shared.R]: I had that before adding encode. I'll make a note to test this before removing but I think we can move ahead with the pr and I will address this later.

File: NAMESPACE [hunk 4]
```diff
@@ -17,6 +17,6 @@ export(drop_read_csv)
 export(drop_search)
 export(drop_share)
 export(drop_upload)
+export(list_shared_links)
```
Comment (ClaytonJY) [inline NAMESPACE]: re-roxygenize

**Pull request merged**